src/sleeplab_format/extractor/preprocess.py

```diff
@@ -277,11 +277,9 @@ def upsample_linear(
     """Linear interpolation for upsampling signals such as SpO2."""
     fs_orig = attributes.sampling_rate
     n = len(s)
-    int_factor = fs_new // fs_orig
-    x = np.arange(0, int_factor*n, int_factor)
-    x_new = np.arange(int_factor*n - 1)
+    n_new = int(fs_new * n/fs_orig) #length of the upsampled signal (must be integer)
+    x = np.linspace(0, n-1, n)
```

Comment:
This probably should be
```python
x = np.linspace(0, n_new-1, n)
``` 
since we want the end points to match. Now the code just repeats the last value of the original signal to the new length.

Reply:
Ahhh yes, that is true. Did not realize that before.